R/verb-copy-to.R

```diff
@@ -227,11 +227,25 @@ sql_values_clause <- function(con, df, row = FALSE, lvl = 0) {
   rows <- rlang::exec(paste, !!!escaped_values, sep = ", ")
   rows_sql <- sql(paste0(if (row) "ROW", "(", rows, ")"))
 
-  union_query <- set_op_query(
-    null_row_query,
-    sql_format_clauses(list(sql_clause("VALUES", rows_sql)), lvl = lvl + 1, con = con),
-    type = "UNION ALL",
-  )
+  rows_clauses <- list(sql_clause("VALUES", rows_sql))
+
+  if (derived) {
+    rows_query <- sql_format_clauses(rows_clauses, lvl = 0, con = con)
+
+    derived_sql <- sql(paste0("drvd(", escape(ident(colnames(df)), con = con), ")"))
+
+    rows_query <- sql_query_select(
+      con,
+      sql("*"),
+      # sql_subquery() can't use sql() or ident_q() as `name` argument
```

Comment:
It is quite annoying that `ident(ident(x))` actually does the quoting twice. Might be worth trying to fix this at some point...

Reply:
`ident(ident_q(x))` or `ident(sql(x))` are problematic. Can we really fix this without breaking anything?

Reply:
I'm not sure. But I don't see where I actually want `ident()` to overwrite the `ident_q` or `sql` class. So, this feels more like a common source for errors than for intended behaviour. But this is something for a different PR.